File: typescript/src/workshop/m5/s05_extractsubclass/step4/Programme.ts
import { Screening } from './Screening.js';

/** Krok 4: bez zmian - klient przeżył całą ekstrakcję bez modyfikacji od kroku 1. */
export class Programme {
  line(title: string, format: string, guest: string | null): string {
    const screening: Screening = guest === null
      ? Screening.regular(title, format)
      : Screening.premiere(title, format, guest);
    return screening.describe() + ' | ' + screening.price();
  }
}

File: csharp/src/Training.Workshop/M5/S05ExtractSubclass/Step4/Programme.cs
namespace Training.Workshop.M5.S05ExtractSubclass.Step4;

/// <summary>Krok 4: bez zmian - klient przeżył całą ekstrakcję bez modyfikacji od kroku 1.</summary>
public sealed class Programme
{
    public string Line(string title, string format, string? guest)
    {
        var screening = guest == null
            ? Screening.Regular(title, format)
            : Screening.Premiere(title, format, guest);
        return screening.Describe() + " | " + screening.Price();
    }
}
